adstb/app.py

```diff
@@ -294,45 +301,38 @@ def _retrieve_abstract_template(self, url):
         msg = TurboBeeMsg(target=url)
         i = 0
         
-        parts = self._parse_bbb_url(url)
-        
         while not self.harvest_webpage(msg) and i < 3:
             self.logger.warn('Retrying to fetch: ' + url)
             i += 1
-        
+
         html = msg.get_value()
         html = html.decode('utf8')
-        
+
+
         # some basic checks
-        if 'data-widget="ShowAbstract"' not in html:
-            raise Exception("Failed to fetch a valid html page for: %s" % url)
-        
+        # if url not in html or 'data-widget="ShowAbstract"' not in html:
+        # I think the entire url string is not in html, need to re-evaluate
+        if 'data-widget="ShowAbstract"' not in html:            
+            raise Exception('Cannot process fetched page, or data-widget for {}'.format(url))
+        # TODO; find the sections and replace them with symbolic names {tags}, {abstract}....
         x = html.find('data-highwire')
         while x > 0:
             x -= 1
             if html[x] == '<':
                 break
         
-        end = html.find('data-highwire', x)
-        while html.find('data-highwire', end+1) > 0:
-            end = html.find('data-highwire', end+1)
-
-        
-        while html[end] != '>':
-            end += 1
-        
-            
+        end = html.find('</head')
```

Comment:
this is likely unsafe; you may be removing too much stuff (if you remove something important, bbb will be broken)

Reply:
I don't think I added this, perhaps we should talk about what you'd like instead.

Reply:
i think you did, i'm looking at your changes ;-)

my comment is about line 318; if we search for data-highwire, we get the beginning of the tags section -- but there can be lots of stuff between that and the end of the `</head>`

so you are potentially replacing too much; was there a problem with the original lines 317-322?

Reply:
thx, reverted code back to original